accord-core/src/main/java/accord/local/Node.java

```diff
@@ -550,6 +582,8 @@ public RoutingKey selectRandomHomeKey(TxnId txnId)
         // TODO (expected): should we try to pick keys in the same Keyspace in C*? Might want to adapt this to an Agent behaviour
         if (ranges.isEmpty()) // should not really happen, but pick some other replica to serve as home key
             ranges = topology().globalForEpoch(txnId.epoch()).ranges();
+        if (ranges.isEmpty())
```

Comment:
not 100% required in this patch, but found it failed in my patch and rather than revert this one change, just pulled everything forward...